src/regex_redux.rs

```diff
@@ -9,32 +9,32 @@
 
 extern crate regex;
 
+use regex::bytes::Regex;
 use std::borrow::Cow;
+use std::error::Error;
 use std::fs;
 use std::sync::Arc;
 use std::thread;
 
-macro_rules! regex { ($re:expr) => { ::regex::bytes::Regex::new($re).unwrap() } }
-
-fn main() {
-    let mut seq = fs::read("/dev/stdin").unwrap();
+fn main() -> Result<(), Box<Error>> {
+    let mut seq = fs::read("/dev/stdin")?;
     let ilen = seq.len();
 
     // Remove headers and newlines.
-    seq = regex!(">[^\n]*\n|\n").replace_all(&seq, &b""[..]).into_owned();
+    seq = Regex::new(">[^\n]*\n|\n")?.replace_all(&seq, &b""[..]).into_owned();
```

Comment:
I think an `unwrap` on a regex built from a literal string in the source text is idiomatic, since a compilation error there is a bug.